api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto

```diff
@@ -35,6 +44,22 @@ message GcpAuthnFilterConfig {
   // Request header location to extract the token. By default (i.e. if this field is not specified), the token
   // is extracted to the Authorization HTTP header, in the format "Authorization: Bearer <token>".
   TokenHeader token_header = 4;
+
+  // Config for where to retrieve the audience URL of the receiving service that
+  // performs token authentication.
+  // Defaults to CLUSTER_METADATA.
+  AudienceProvider audience_provider = 5 [(validate.rules).enum = {defined_only: true}];
+
+  // Setting this is required when the audience provider is
+  // :ref:`REQUEST_HEADER<envoy_v3_api_enum_value_extensions.filters.http.gcp_authn.v3.GcpAuthnFilterConfig.AudienceProvider.REQUEST_HEADER>`.
+  //
+  // The value of this request header will be used as the audience URL of the
+  // receiving service.
+  //
+  // .. note::
+  //
+  //   If the header appears multiple times only the first value is used.
+  string audience_header = 6;
```

Comment:
It would be better to express that as a oneof. The oneof can default to the existing behavior and then add new behavior for the header option.

Reply:
@mattklein123 what do you think about the above proposed changes? If they look reasonable then I'll move forward with making the code changes